test/e2e/user_defined_sa_test.go

```diff
@@ -0,0 +1,334 @@
+	require.NoError(t, err)
+
+	conditionGot := mustHaveCondition(t, ipGot, v1alpha1.InstallPlanInstalled)
+	assert.Equal(t, corev1.ConditionFalse, conditionGot.Status)
+	assert.Equal(t, v1alpha1.InstallPlanReasonComponentFailed, conditionGot.Reason)
+	assert.Contains(t, conditionGot.Message, fmt.Sprintf("is forbidden: User \"system:serviceaccount:%s:%s\" cannot create resource", namespace, saName))
+
+	// Verify that all step resources are in Unknown state.
+	for _, step := range ipGot.Status.Plan {
+		assert.Equal(t, v1alpha1.StepStatusUnknown, step.Status)
+	}
+}
+
+func TestUserDefinedServiceAccountWithPermission(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, false)
+
+	// Create the CatalogSource
+	kubeclient := newKubeClient(t)
+	crclient := newCRClient(t)
+
+	namespace := genName("scoped-ns-")
+	_, ceanupNS := newNamespace(t, kubeclient, namespace)
+	defer ceanupNS()
+
+	// Create a service account, add enough permission to it so that operator install is successful.
+	saName := genName("scoped-sa")
+	_, cleanupSA := newServiceAccount(t, kubeclient, namespace, saName)
+	defer cleanupSA()
+	cleanupPerm := grantPermission(t, kubeclient, namespace, saName)
+	defer cleanupPerm()
+
+	// Add an OperatorGroup and specify the service account.
+	ogName := genName("scoped-og-")
+	_, cleanupOG := newOperatorGroupWithServiceAccount(t, crclient, namespace, ogName, saName)
+	defer cleanupOG()
+
+	catsrc, subSpec, catsrcCleanup := newCatalogSource(t, kubeclient, crclient, namespace)
+	defer catsrcCleanup()
+
+	// Ensure that the catalog source is resolved before we create a subscription.
+	_, err := fetchCatalogSource(t, crclient, catsrc.GetName(), namespace, catalogSourceRegistryPodSynced)
+	require.NoError(t, err)
+
+	subscriptionName := genName("scoped-sub-")
+	cleanupSubscription := createSubscriptionForCatalog(t, crclient, namespace, subscriptionName, catsrc.GetName(), subSpec.Package, subSpec.Channel, subSpec.StartingCSV, subSpec.InstallPlanApproval)
+	defer cleanupSubscription()
+
+	// Wait until an install plan is created.
+	subscription, err := fetchSubscription(t, crclient, namespace, subscriptionName, subscriptionHasInstallPlanChecker)
+	require.NoError(t, err)
+	require.NotNil(t, subscription)
+
+	// We expect the InstallPlan to be in status: Complete.
+	ipName := subscription.Status.Install.Name
+	ipPhaseCheckerFunc := buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete)
+	ipGot, err := fetchInstallPlanWithNamespace(t, crclient, ipName, namespace, ipPhaseCheckerFunc)
+	require.NoError(t, err)
+
+	conditionGot := mustHaveCondition(t, ipGot, v1alpha1.InstallPlanInstalled)
+	assert.Equal(t, v1alpha1.InstallPlanConditionReason(""), conditionGot.Reason)
+	assert.Equal(t, corev1.ConditionTrue, conditionGot.Status)
+	assert.Equal(t, "", conditionGot.Message)
+
+	// Verify that all step resources are in Created state.
+	for _, step := range ipGot.Status.Plan {
+		assert.Equal(t, v1alpha1.StepStatusCreated, step.Status)
+	}
```

Comment:
We should probably also re-kick installplans in a failed state if the operator serviceaccount changes (doesn't need to be this pr, just a note for our backlog)

Reply:
Added a jira ticket for this - https://jira.coreos.com/browse/OLM-1118